src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -821,6 +821,14 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                 CollectTupleFieldMemberNames(name, i + 1, numElements, ref elementNames);
 
                 BoundExpression boundArgument = BindValue(argumentSyntax.Expression, diagnostics, BindValueKind.RValue);
+                if (boundArgument.Type?.SpecialType == SpecialType.System_Void)
```

Comment:
>if (boundArgument.Type?.SpecialType == SpecialType.System_Void) [](start = 16, length = 63)

It Is unfortunate that BindValueKind.RValue doesn't error on void. Because of that every consumer of BindValue API has to check for it explicitly.